Expose `quinn::EndpointConfig` for `{Client,Server}Config`

pablosichert opened the pull request:
This PR exposes the `quinn::EndpointConfig` via `quic_endpoint_config`/`quic_endpoint_config_mut` on `ClientConfig`/`ServerConfig`, analogously to how `quinn::{Client,Server}` are currently exposed.

To give a bit of context to the use case: I'd like to manually manage connection IDs on the application level and would like to set the [`EndpointConfig::cid_generator`](https://github.com/quinn-rs/quinn/blob/ead9b9316c155073c0984a243aeb9b84c5465298/quinn-proto/src/config.rs#L651-L665) for that purpose.

If you have another proposal on how to expose this or don't like to expose it all, let me know. I just figured opening this PR was as quick as opening a discussion on what I'd like to change.

BiagioFesta reviewed (changes requested):
Can you please squash those 3 commit into a single one? I think we can aggregate them

pablosichert commented:
Sure, done. I usually push commits to a feature branch since that makes it easier for reviewers to see what I changed since last time. And then use the "squash and merge" option on GitHub. Anway, squashed these now!
